pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -59,68 +59,71 @@ type GoRoutineMap interface {
 	Wait()
 }
 
-// NewGoRoutineMap returns a new instance of GoRoutineMap.
-func NewGoRoutineMap(exponentialBackOffOnError bool) GoRoutineMap {
+// NewGoroutineMap returns a new instance of GoroutineMap.
+func NewGoroutineMap(exponentialBackOffOnError bool) GoroutineMap {
 	g := &goRoutineMap{
 		operations:                make(map[string]operation),
 		exponentialBackOffOnError: exponentialBackOffOnError,
+		lock: &sync.Mutex{},
 	}
-	g.cond = sync.NewCond(g)
+
+	g.cond = sync.NewCond(g.lock)
 	return g
 }
 
 type goRoutineMap struct {
 	operations                map[string]operation
 	exponentialBackOffOnError bool
 	cond                      *sync.Cond
-	sync.Mutex
+	lock                      *sync.Mutex
```

Comment:
+1

On Tue, Jul 19, 2016 at 2:31 PM, Saad Ali <notifications@github.com> wrote:

> In pkg/util/goroutinemap/goroutinemap.go
> <https://github.com/kubernetes/kubernetes/pull/28939#discussion_r71427500>
> :
>
> >  	return g
> >  }
> >
> >  type goRoutineMap struct {
> >  	operations                map[string]operation
> >  	exponentialBackOffOnError bool
> >  	cond                      *sync.Cond
> > -	sync.Mutex
> > +	lock                      *sync.Mutex
>
> To address clayton's earlier comment:
>
> nit: we prefer lock sync.Mutex because it prevents Mutex from being public
> to the goRoutineMap, unless you have a need for that.
> I would prefer getting access to Lock being impossible from outside this
> object without a good justification.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28939/files/f344f35d4d1b693130ed5f82f76696c9064b2609#r71427500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASzXe3rRsIQCYlsd5goLyTSUR5XsiCrJks5qXUIngaJpZM4JMGj8>
> .
>